src/key.rs

```diff
@@ -612,20 +766,38 @@ impl KeyPair {
         }
     }
 
-    /// Serialize the key pair as a secret key byte value
+    /// Serializes the key pair as a secret key byte value.
     #[inline]
     pub fn serialize_secret(&self) -> [u8; constants::SECRET_KEY_SIZE] {
         *SecretKey::from_keypair(self).as_ref()
     }
 
-    /// Tweak a keypair by adding the given tweak to the secret key and updating the public key
+    /// Tweaks a keypair by adding the given tweak to the secret key and updating the public key
     /// accordingly.
     ///
-    /// Will return an error if the resulting key would be invalid or if the tweak was not a 32-byte
+    /// # Errors
+    ///
+    /// Returns an error if the resulting key would be invalid or if the tweak was not a 32-byte
     /// length slice.
     ///
     /// NB: Will not error if the tweaked public key has an odd value and can't be used for
     ///     BIP 340-342 purposes.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(feature="rand")] {
+    /// use secp256k1::{rand, Secp256k1, KeyPair};
+    /// use secp256k1::rand::{RngCore, thread_rng};
+    ///
+    /// let secp = Secp256k1::new();
+    /// let mut tweak = [0u8; 32];
+    /// thread_rng().fill_bytes(&mut tweak);
+    ///
+    /// let mut key_pair = KeyPair::new(&secp, &mut thread_rng());
+    /// key_pair.tweak_add_assign(&secp, &tweak).expect("Tweak error");
```

Comment:
I don't think `"Tweak error"` is any more helpful than just an unwrap here.
so I think it should either be `unwrap` or  `"Improbable to fail with a randomly generated tweak"`

Reply:
Used the descriptive string suggested, thanks.